src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs

```diff
@@ -44,18 +44,28 @@ public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, strin
         _transport = transport;
         Name = name;
         Port = port;
-
         TargetPort = targetPort;
+        IsExternal = isExternal ?? false;
+        IsProxied = isProxied;
 
-        // If the target port was not explicitly set and the service is not being proxied,
-        // we can set the target port to the port.
-        if (TargetPort is null && !isProxied)
+        if (!isProxied)
         {
-            TargetPort = port;
+            // For proxy-less Endpoints the client port and target port should be the same.
+            // Note that this is just a "sensible default"--the consumer of the EndpointAnnotation is free
```

Comment:
The API for creating an EndpointAnnotation makes this difficult since the annotation is not returned.

Reply:
You can use this overload https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting/ResourceBuilderExtensions.cs#L366 to set individual EndpointAnnotation properties, but I agree it is not super discoverable.